pkg/node/manager/manager.go

```diff
@@ -132,12 +139,21 @@ func (m *manager) AddNode(nodeName string) error {
 	var newNode node.Node
 	var nodeFound bool
 
-	newNode, nodeFound = m.dataStore[k8sNode.Name]
+	_, nodeFound = m.dataStore[k8sNode.Name]
 	if nodeFound {
 		log.Info("node is already processed, not processing add event again")
 		return nil
 	}
 
+	if err = m.CreateCNINodeIfNotExisting(k8sNode); err != nil {
+		m.Log.Error(err, "Failed to create CNINode for k8sNode", "NodeName", k8sNode.Name)
+		// we shouldn't see AlreadyExists error here
+		// double-check here to avoid retry on this error
```

Comment:
I'd be a bit cautious about "double checking". The code should work as designed, and not need double checking.

Reply:
Removed it. You are right. It should work. The check is redundant.